client/dashboard/sites/settings-static-file-404/index.tsx

```diff
@@ -0,0 +1,155 @@
+import { settings } from '@wordpress/icons';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState } from 'react';
+import { siteQuery, siteStaticFile404Query, siteStaticFile404Mutation } from '../../app/queries';
+import { Callout } from '../../components/callout';
+import PageLayout from '../../components/page-layout';
+import SettingsPageHeader from '../settings-page-header';
+import calloutIllustrationUrl from './callout-illustration.svg';
+import type { Site } from '../../data/types';
+import type { Field } from '@automattic/dataviews';
+
+export function canSetStaticFile404Handling( site: Site ) {
+	return site.is_wpcom_atomic;
+}
+
+const fields: Field< { setting: string } >[] = [
+	{
+		id: 'setting',
+		label: __( 'Response' ),
+		Edit: 'select',
+		description: __(
+			'Assets are images, fonts, JavaScript, and CSS files that web browsers request as part of loading a web page. This setting controls how the web server handles requests for missing asset files.'
+		),
+		elements: [
+			{ value: 'default', label: __( 'Default' ) },
+			{ value: 'lightweight', label: __( 'Send a lightweight File-Not-Found page' ) },
+			{ value: 'wordpress', label: __( 'Delegate request to WordPress' ) },
+		],
+	},
+];
+
+const form = {
+	type: 'regular' as const,
+	fields: [ 'setting' ],
+};
+
+export default function SiteStaticFile404Settings( { siteSlug }: { siteSlug: string } ) {
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const { data: site } = useQuery( siteQuery( siteSlug ) );
+	const { data: setting } = useQuery( {
+		...siteStaticFile404Query( siteSlug ),
+		enabled: site && canSetStaticFile404Handling( site ),
+	} );
+	const mutation = useMutation( siteStaticFile404Mutation( siteSlug ) );
+
+	const [ formData, setFormData ] = useState< { setting: string } >( {
+		setting: setting ?? '',
+	} );
+
+	if ( ! site ) {
+		return null;
+	}
+
+	const isDirty = formData.setting !== setting;
+	const { isPending } = mutation;
+
+	const handleUpgradePlan = () => {
+		const params = new URLSearchParams();
+		params.append( 'redirect_to', window.location.href.replace( window.location.origin, '' ) );
+
+		window.location.href = `/checkout/${ encodeURIComponent(
+			siteSlug
+		) }/business?${ params.toString() }`;
+	};
+
+	const handleSubmit = ( e: React.FormEvent ) => {
+		e.preventDefault();
+		if ( ! canSetStaticFile404Handling( site ) ) {
+			return;
+		}
+
+		mutation.mutate( formData.setting, {
+			onSuccess: () => {
+				createSuccessNotice( __( 'Settings saved.' ), { type: 'snackbar' } );
+			},
+			onError: () => {
+				createErrorNotice( __( 'Failed to save settings.' ), {
+					type: 'snackbar',
+				} );
+			},
+		} );
+	};
+
+	return (
+		<PageLayout
+			size="small"
+			header={ <SettingsPageHeader title={ __( 'Handling requests for nonexistent assets' ) } /> }
+		>
+			{ ! canSetStaticFile404Handling( site ) ? (
+				<Callout
+					icon={ settings }
+					headingLevel={ 4 }
```

Comment:
@p-jackson seems that in the Callout component, the class name `dashboard-callout__title` is setting the heading to 15px, which is as per design. However, the default `headingLevel = 6` sometimes overrides this with Heading's inline CSS.

Reply:
The default `headingLevel` is pretty much arbitrary. I just wanted to make sure it was semantically a heading. I'm not sure what the best way to do typography is in the new DS. I feel we probably want to use `Text` and `Heading` components, but perhaps this is not true? If we're hoping to not rely on so much custom CSS, then I don't see how else to do it other than having components that implement our standard typography styles.

Short story: feel free to make the default `headingLevel` 4 if that has less issues with style overriding. Given the places the `Callout` component will usually appear, `<h4>` should probably be fine.

Reply:
In the development environment, we run into the usual issue where the loading order differs sometimes between the CSS having the higher precedence (dashboard-callout__title), while other times the inline CSS of Heading takes precedence. Not sure if it's an issue on staging/prod, but in the meantime h4 is close enough (16.25px) to 15px.